tests/utils/TestHelper.ts

```diff
@@ -0,0 +1,252 @@
+import Str from 'expensify-common/lib/str';
+import Onyx from 'react-native-onyx';
+import CONST from '@src/CONST';
+import * as Session from '@src/libs/actions/Session';
+import HttpUtils from '@src/libs/HttpUtils';
+import * as NumberUtils from '@src/libs/NumberUtils';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {PersonalDetails, Report} from '@src/types/onyx';
+import waitForBatchedUpdates from './waitForBatchedUpdates';
+
+type MockFetch = ReturnType<typeof jest.fn> & {
+    pause?: () => void;
+
+    fail?: () => void;
+
+    succeed?: () => void;
+
+    resume?: () => Promise<void>;
+};
+
+type Response = {ok: boolean; json: () => Promise<{jsonCode: number}>};
+type QueueItem = (value: Response | PromiseLike<Response>) => void;
+
+type FormData = {
+    entries: () => Array<[string, string | Blob]>;
+};
+
+type Listener = () => void;
+
+function buildPersonalDetails(login: string, accountID: number, firstName = 'Test'): PersonalDetails {
+    return {
+        accountID,
+        login,
+        avatar: 'https://d2k5nsl2zxldvw.cloudfront.net/images/avatars/avatar_7.png',
+        avatarThumbnail: 'https://d2k5nsl2zxldvw.cloudfront.net/images/avatars/avatar_7.png',
+        displayName: `${firstName} User`,
+        firstName,
+        lastName: 'User',
+        pronouns: '',
+        timezone: CONST.DEFAULT_TIME_ZONE,
+        phoneNumber: '',
+    };
+}
+
+/**
+ * Simulate signing in and make sure all API calls in this flow succeed. Every time we add
+ * a mockImplementationOnce() we are altering what Network.post() will return.
+ */
+function signInWithTestUser(accountID = 1, login = 'test@user.com', password = 'Password1', authToken = 'asdfqwerty', firstName = 'Test') {
+    const originalXhr = HttpUtils.xhr;
+
+    HttpUtils.xhr = jest.fn().mockImplementation(() => {
+        // Your mocked response object
+        const mockedResponse = {
```

Comment:
```suggestion
        const mockedResponse: Response = {
```

Could we type this object?